flask-homework/app4.py

```diff
@@ -0,0 +1,234 @@
+#
+#     # Add users and books to the database
+#     db.session.add_all([user1, user2, user3, user4, user5, user6, user7, user8, book1, book2, book3, book4, book5, book6, book7, book8])
+#     db.session.commit()
+#
+#     # Now that the users and books are saved in the database, you can add purchases with the correct user_id and book_id
+#     purchase1 = Purchase(user_id=user1.id, book_id=book1.id, purchase_date=datetime.now())
+#     purchase2 = Purchase(user_id=user2.id, book_id=book2.id, purchase_date=datetime.now())
+#     purchase3 = Purchase(user_id=user3.id, book_id=book3.id, purchase_date=datetime.now())
+#     purchase4 = Purchase(user_id=user4.id, book_id=book4.id, purchase_date=datetime.now())
+#     purchase5 = Purchase(user_id=user5.id, book_id=book5.id, purchase_date=datetime.now())
+#     purchase6 = Purchase(user_id=user6.id, book_id=book6.id, purchase_date=datetime.now())
+#     purchase7 = Purchase(user_id=user7.id, book_id=book7.id, purchase_date=datetime.now())
+#     purchase8 = Purchase(user_id=user8.id, book_id=book8.id, purchase_date=datetime.now())
+#
+#     # Adding purchases to the database
+#     db.session.add_all([purchase1, purchase2, purchase3, purchase4, purchase5, purchase6, purchase7, purchase8])
+#     db.session.commit()
+
+
+# Models
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(100), nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
+    purchases = db.relationship('Purchase', back_populates='user')
+
+    def serialize(self):
+        return {
+            'id': self.id,
+            'name': self.name,
+            'email': self.email,
+        }
+
+
+class Book(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(200), nullable=False)
+    author = db.Column(db.String(100), nullable=False)
+    price = db.Column(db.Float, nullable=False)
+    purchases = db.relationship('Purchase', back_populates='book')
+
+    def serialize(self):
+        return {
+            'id': self.id,
+            'title': self.title,
+            'author': self.author,
+            'price': self.price,
+        }
+
+
+class Purchase(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
+    book_id = db.Column(db.Integer, db.ForeignKey('book.id'), nullable=False)
+    purchase_date = db.Column(db.DateTime, nullable=False)
+
+    user = db.relationship('User', back_populates='purchases')
+    book = db.relationship('Book', back_populates='purchases')
+
+    def serialize(self):
+        return {
+            'id': self.id,
+            'user_id': self.user_id,
+            'book_id': self.book_id,
+            'purchase_date': self.purchase_date.isoformat(),
+        }
+
+
+# Endpoints
+# Exercise 5-6
+@app.route('/users', methods=['GET'])
```

Comment:
nit: if only one method is going to use in view, easier
```suggestion
@app.get('/users')
```